ap_src/ap_app/utils/teams_utils.py

```diff
@@ -0,0 +1,44 @@
+# Third-party Libraries
+import hashlib
+import requests
+
+# Environment
+import environ
+
+env = environ.Env()
+environ.Env.read_env()
+
+
+def retrieve_calendar_data(user, sortValue):
+    data = None
+    r = None
+
+    try:
+        token = (str(user) + "AbsencePlanner").encode()
+        encryption = hashlib.sha256(token).hexdigest()
+        r = requests.get(env("TEAM_DATA_URL") + "api/user/teams/?format=json&username={}&sort={}".format(user.username, sortValue), headers={"TEAMS-TOKEN": encryption})
+    except:
+        print("API Failed to connect")
+    
+    if r is not None and r.status_code == 200:
+        data = r.json()
+    
+    return data
+
+def get_users_sharing_teams(current_user, user_model):
+    teams = retrieve_calendar_data(user_model, None)
+    if teams is None or teams == []: # The current_user is not in any teams
+        return {} # Avoid error from iterating through None type
+
+    users_sharing_teams = []
+    for team in teams:
+        team = team["team"]
+        for member in team["members"]:
+            username = member["user"]["username"]
+            users_sharing_teams.append(username)
+    users_sharing_teams = set(users_sharing_teams)
+    users_sharing_teams.remove(current_user)
```

Comment:
Consider what would happen here if we try to remove current_user, but it doesn't exist in users_sharing_teams. I'm assuming this should never happen but it's still worth having error handling when trying to do something like this.

Reply:
Yes I was aware of this issue. The function here removes the `current_user` from the list of members sharing teams with them because the user cannot share a team with themselves. It should not ever cause an error as the many to many relationship for permission to switch with themselves is always made when a new account is created. However, I suppose it would be a sensible idea to implement error handling here in case something is severely broken.

Reply:
Fixed with ccaa924